plaid/src/apis/okta/mod.rs

```diff
@@ -30,15 +41,103 @@ pub struct Okta {
+    GetUserInfo,
+    RemoveUserFromGroup
+}
+
+impl OktaOperation {
+    /// Get the correct Okta OAuth 2.0 scope, depending on which operation needs to be performed.
+    pub fn to_okta_scope(&self) -> &str {
+        match self {
+            // https://developer.okta.com/docs/api/openapi/okta-management/management/tag/User/#tag/User/operation/getUser
+            Self::GetUserInfo => "okta.users.read",
+            // https://developer.okta.com/docs/api/openapi/okta-management/management/tag/Group/#tag/Group/operation/unassignUserFromGroup
+            Self::RemoveUserFromGroup => "okta.groups.manage"
+        }
+    }
 }
 
 impl Okta {
     pub fn new(config: OktaConfig) -> Self {
         let client = reqwest::Client::builder()
-            .timeout(Duration::from_secs(config.api_timeout_seconds))
+            .timeout(std::time::Duration::from_secs(config.api_timeout_seconds))
             .build()
             .unwrap();
 
         Self { config, client }
     }
+
+    /// Return an appropriate authorization header, to be used when making a REST call.
+    pub async fn get_authorization_header(&self, op: &OktaOperation) -> Result<String, OktaError> {
+        match &self.config.authentication {
+            Authentication::ApiKey { token } => Ok(format!("SSWS {}", token)),
+            Authentication::OktaApp { .. } => {
+                let access_token = self.get_access_token(op).await?;
+                Ok(format!("Bearer {}", access_token))
+            }
+        }
+    }
+
+    /// Construct a JWT signed with Plaid's private key, to be later exchanged for an access token.
+    fn get_jwt(&self) -> Result<String, OktaError> {
+        match &self.config.authentication {
+            Authentication::ApiKey { token: _ } => panic!("Trying to get a JWT but this is not configured as an Okta app. This should be impossible."),
+            Authentication::OktaApp { client_id, private_key } => {
+                let key_pair = RS256KeyPair::from_pem(&private_key).map_err(|_| OktaError::BadPrivateKey)?;
+                // For more details, see https://developer.okta.com/docs/guides/implement-oauth-for-okta-serviceapp/main/#create-and-sign-the-jwt
+                let claims = Claims::create(Duration::from_secs(30))
+                    .with_issuer(client_id)
+                    .with_audience(format!("https://{}/oauth2/v1/token", self.config.domain))
+                    .with_subject(client_id);
+                let jwt = key_pair.sign(claims).map_err(|_| OktaError::JwtSignatureFailure)?;
+                Ok(jwt)
+            }
+        }
+    }
+
+    /// Obtain an access token from Okta, in exchange for a properly constructed JWT.
+    async fn get_access_token(&self, op: &OktaOperation) -> Result<String, OktaError> {
+        match &self.config.authentication {
+            Authentication::ApiKey { token: _ } => panic!("Trying to get an access token but this is not configured as an Okta app. This should be impossible."),
```

Comment:
Same ask as above, generally we never want the Plaid runtime to crash so anytime we use panic!, expect, or unwrap, we need to justify it's usage as to why it will never happen (once start up has finished).

Generally easier if we can guarantee it won't happen (which isn't always possible) but advisable.

Reply:
Fixed by rearranging the code